gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpec.java

```diff
@@ -75,13 +78,22 @@ public class FlowSpec implements Configurable, Spec {
   /** Human-readable description of the flow spec */
   final String description;
 
-  /** Flow config as a typesafe config object */
-  final Config config;
+  /* Note that since getConfig() and getConfigAsProperties() are independent accessors, `volatile` doesn't ensure a
+  * consistent view between them. In a multi-threaded scenario one should use the following access mechanism:
+  *    FlowSpec fs = ...
+  *    synchronized (fs) {
+  *      fs.getConfig()
+  *      fs.getConfigAsProperties()
+  *    }
+  */
+
+  /** Flow config as a typesafe config object which can be replaced */
+  volatile Config config;
```

Comment:
Probably a sign that we should reduce the API if it's being used like this, but currently I can't think of a scenario where we have concurrent read or updates of the same flowspec.

Reply:
I agree, I don't really expect a scenario with concurrent reads/updates so I am marking it as volatile to indicate I'm aware there could be a synchronization issue but don't want to over engineer a solution.

Reply:
package-protected `volatile` is atypical.  I like the `private volatile` + accessor that arjun suggested.